src/Repository/Storable/Introduction.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Friendica\Repository\Storable;
+
+use Friendica\Collection;
+use Friendica\Core\Protocol;
+use Friendica\Model;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\ActivityPub;
+use Friendica\Protocol\Diaspora;
+use Friendica\Repository\Storable;
+use Friendica\Util\DateTimeFormat;
+
+/**
+ * @method Model\Introduction       selectFirst(array $condition)
+ * @method Collection\Introductions select(array $condition = [], array $params = [])
+ * @method Collection\Introductions selectByBoundaries(array $condition = [], array $params = [], int $max_id = null, int $since_id = null, int $limit = self::LIMIT)
+ */
+class Introduction extends Storable
+{
+	protected static $table_name = 'intro';
+
+	protected static $model_class = Model\Introduction::class;
+
+	protected static $collection_class = Collection\Introductions::class;
+
+	/**
+	 * Confirms a follow request and sends a notic to the remote contact.
+	 *
+	 * @param Model\Introduction $Introduction
+	 * @param bool               $duplex       Is it a follow back?
+	 * @param bool|null          $hidden       Should this contact be hidden? null = no change
+	 * @return bool
+	 * @throws HTTPException\InternalServerErrorException
+	 * @throws HTTPException\NotFoundException
+	 * @throws \ImagickException
+	 */
+	public function confirm(Model\Introduction $Introduction, bool $duplex = false, bool $hidden = null)
+	{
+		$this->logger->info('Confirming follower', ['cid' => $this->{'contact-id'}]);
+
+		$contact = Model\Contact::selectFirst([], ['id' => $Introduction->{'contact-id'}, 'uid' => $Introduction->uid]);
+
+		if (!$contact) {
+			throw new HTTPException\NotFoundException('Contact record not found.');
+		}
+
+		$newRelation = $contact['rel'];
+		$writable = $contact['writable'];
+
+		if (!empty($contact['protocol'])) {
+			$protocol = $contact['protocol'];
+		} else {
+			$protocol = $contact['network'];
+		}
+
+		if ($protocol == Protocol::ACTIVITYPUB) {
+			ActivityPub\Transmitter::sendContactAccept($contact['url'], $contact['hub-verify'], $contact['uid']);
+		}
+
+		if (in_array($protocol, [Protocol::DIASPORA, Protocol::ACTIVITYPUB])) {
+			if ($duplex) {
+				$newRelation = Model\Contact::FRIEND;
+			} else {
+				$newRelation = Model\Contact::FOLLOWER;
+			}
+
+			if ($newRelation != Model\Contact::FOLLOWER) {
+				$writable = 1;
+			}
+		}
+
+		$fields = [
+			'name-date' => DateTimeFormat::utcNow(),
+			'uri-date'  => DateTimeFormat::utcNow(),
+			'blocked'   => false,
+			'pending'   => false,
+			'protocol'  => $protocol,
+			'writable'  => $writable,
+			'hidden'    => $hidden ?? $contact['hidden'],
+			'rel'       => $newRelation,
+		];
+		$this->dba->update('contact', $fields, ['id' => $contact['id']]);
```

Comment:
Here is one example why I don't feel **so** comfortable:
We have to inject a `ContactRepository` inside a `IntroductionRepository`.. Would it be possible to have it the other way too? Like a `ContactRepository->confirm()` updates an `Introduction` .. or more bad: There is a model in between and we're not aware of it.

=> This leads to dependency-circles:
IntrocutionRepository => ContactRepository => IntroductionRepository => ... which would cause all Dependency Injection libraries to brick

If we would have a `confirm()` method inside another class, we cut this circle
- Introduction->confirm() => IntroductionRepository
- Introduction->confirm() => ContactRepository

I found a clear description of this problem in a book I read.. But I've to search for it and try to reference it ..

Reply:
This is what I did before and you told me it was no good! I don't get it!

Reply:
I'm really sorry.. I think it isn't that easy to really find a good solution to takle down the potential problems :-( :-( .. 

What I meant with the static factories are the static `::create()` methods, not the business logic of the model(s) itself :-( .. So the `Repository` and `Factory` classes are mainly for create/update/delete instances, not for the business logic..

The business logic was inside the `Model` classes itself, where I said that I don't want to overload the class itself .. But this was a different discussion than the static factory problem ...

It's the "Static vs. dynamic factory" topic here: https://friendica.philipp.info/display/f1417ed3-155e-05fd-70b6-347014799460
not the "Model class" topic...

Reply:
Mmmkay', I'll see what I can do tonight to reconcile our points of view. You'll tell me what you think of it.

Reply:
I think we're almost there, having a common line ...